### DISCUSSION TITLE
Kyuubi & Superset Integration

### DISCUSSION THREAD

**Discussion by mmuru** [apache/kyuubi#2145] [Q&A] [answered]
I tried to connect Kyuubi from Apache Superset without any success so  I need additional information. I am using Kyuubi custom build from Master branch. I followed Superset [connection](https://superset.apache.org/docs/databases/installing-database-drivers) driver and SQLAlchemy dialect and tried below options 

1. Using Kyuubi mysql protocol:  mysql://root:kyuubi@192.168.1.69:3309/default  - able to reach Kyuubi server but Superset throws 

```
[SupersetError(message='(MySQLdb._exceptions.ProgrammingError) (2014, "Commands out of sync; you can\'t run this command now")\n(Background on this error at: http://sqlalche.me/e/13/f405)', error_type=<SupersetErrorType.GENERIC_DB_ENGINE_ERROR: 'GENERIC_DB_ENGINE_ERROR'>, level=<ErrorLevel.ERROR: 'error'>, extra={'engine_name': 'MySQL', 'issue_codes': [{'code': 1002, 'message': 'Issue 1002 - The database returned an unexpected error.'}]})]
```
No errors on the Kyuubi server side. It could be related to Superset side unable to process Kyuubi response. 

2. Using Kyuubi THRIFT_BINARY protocol: I tried different SQLAlchemy dialect  
hive://192.168.1.69:10009/default, hive2://192.168.1.69:10009/default, jdbc+hive2://muru:kyuubi@192.168.1.69:10009/default, jdbc+hive://muru:kyuubi@192.168.1.69:10009/default but none of them are working.

Please, let me know what is the correct SQLAlchemy dialect and also point me to right direction/documentation.  

Thanks in advance.

**Comment by mmuru** [comment, marked as answer]
I made it work with Superset.
````
pip install pyhive[hive]
SQLAlchemy dialect url:  hive://192.168.1.69:10009/default 
````

**Reply by pan3793** [reply]
You got it. BTW, there is a known issue https://github.com/apache/superset/issues/10940#issuecomment-824732323 @SteNicholas is working on it.
